### PR TITLE
Replace device_type in swim module for tagging/un-tagging image with …

### PR DESCRIPTION
**Pull request by Abhishek-121**
…device_image_family_name, add brief description for enum or choices for parameter, changed boolean value from true/false to True/False in all modules.

In this commit -
 
 
-- Remove device_type from the swim_workflow_module and swim_intent and add device_image_family_name attribute for tagging/un-tagging the image as golden in order to avoid confusion.
 
-- For other swim operation only device family is needed which is like "Switches and Hubs" and add the same examples in documentation as well.
 
-- Improve the error log messages for like Site/Device not found in swim workflow module.
 
-- For each parameter in documentation having the choices like device_upgrade_mode( 'install', 'bundle' and 'currentlyExists' ) elaborate these terms in detail for customer ease etc. for other parameter as well.
 
-- For importing the image in Cisco Catalyst Center, two options get changes to remote/local from url/local.
 
-- Updated all the changes in both intent modules and workflow_manager module and updated the respective playbooks as well.

-- Changes all boolean value in modules and inventory workflow playbook from true/false to True/False.

### CODE REVIEW

File: plugins/modules/inventory_workflow_manager.py [hunk 1]
```diff
@@ -1755,9 +1768,9 @@ def get_site_type(self, site_name):
                     site_type = item.get("attributes").get("type")
 
         except Exception as e:
-            self.msg = "Error while fetching the site '{0}'.".format(site_name)
+            self.msg = "Error while fetching the site '{0}' and given site not found in Cisco Catalyst Center".format(site_name)
```
Comment (madhansansel) [inline plugins/modules/inventory_workflow_manager.py:1771]: and the specified site was not found in Cisco Catalyst Center.

File: plugins/modules/swim_intent.py [hunk 2]
```diff
@@ -50,26 +50,26 @@
         type: dict
         suboptions:
           type:
-            description: The source of import, supports url import or local import.
+            description: The source of import, supports remote import or local import.
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:53]: Specifies the import source, supporting local import (file) or remote import (URL)

File: plugins/modules/swim_intent.py [hunk 3]
```diff
@@ -50,26 +50,26 @@
...
             type: str
           local_image_details:
             description: Details of the local path of the image to be imported.
             type: dict
             suboptions:
               file_path:
-                description: File absolute path.
+                description: Give the file absolute path required while importing image from local.
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:60]: description: The absolute path of the file required for local import.
example: "/path/to/your/file"

OR
description: Provide the absolute file path needed to import an image from your local system.

File: plugins/modules/swim_intent.py [hunk 4]
```diff
@@ -50,26 +50,26 @@
...
                 type: str
               is_third_party:
-                description: IsThirdParty query parameter. Third party Image check.
+                description: IsThirdParty query parameter. Third party Image check (Optional).
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:63]: description: Query parameter to determine if the image is from a third party (optional).

File: plugins/modules/swim_intent.py [hunk 5]
```diff
@@ -50,26 +50,26 @@
...
                 type: bool
               third_party_application_type:
-                description: ThirdPartyApplicationType query parameter. Third Party Application Type.
+                description: ThirdPartyApplicationType query parameter. Third Party Application Type (Optional).
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:66]: description: Specify the ThirdPartyApplicationType query parameter to indicate the type of third-party application.(optional)

File: plugins/modules/swim_intent.py [hunk 6]
```diff
@@ -50,26 +50,26 @@
...
                 type: str
               third_party_image_family:
-                description: ThirdPartyImageFamily query parameter. Third Party image family.
+                description: ThirdPartyImageFamily query parameter. Third Party image family (Optional).
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:69]: Provide the ThirdPartyImageFamily query parameter to identify the family of the third-party image. (optional)

File: plugins/modules/swim_intent.py [hunk 7]
```diff
@@ -50,26 +50,26 @@
...
                 type: str
               third_party_vendor:
-                description: ThirdPartyVendor query parameter. Third Party Vendor.
+                description: ThirdPartyVendor query parameter (Optional).
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:72]: Include the ThirdPartyVendor query parameter to specify the vendor of the third party.

File: plugins/modules/swim_intent.py [hunk 8]
```diff
@@ -81,19 +81,20 @@
                 elements: dict
                 suboptions:
                   application_type:
-                    description: Swim Import Via Url's applicationType.
+                    description: Optional parameter indicating the type of application with permitted values(WLC, LINUX, FILREWALL, WINDOWS,
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:84]: FILREWALL spelling mistake...

`An optional parameter that specifies the type of application. Allowed values include WLC, LINUX, FIREWALL, WINDOWS, LOADBALANCER, THIRDPARTY, etc. This is only applicable for third-party image types.`

File: plugins/modules/swim_intent.py [hunk 9]
```diff
@@ -81,19 +81,20 @@
...
+                      LOADBALANCER, THIRDPARTY etc) applicable only for third party image types.
                     type: str
                   image_family:
-                    description: Swim Import Via Url's imageFamily.
+                    description: The name of image family applicable only in case of third party images upload (Optional).
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:88]: Represents the name of the image family and is applicable only when uploading third-party images.

File: plugins/modules/swim_intent.py [hunk 10]
```diff
@@ -81,19 +81,20 @@
...
                     type: str
                   source_url:
-                    description: Swim Import Image Via Url.
+                    description: Required parameter for importing swim image via remote url.
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:91]: A mandatory parameter for importing a SWIM image via a remote URL. This parameter is required when using a URL to import an image.

File: plugins/modules/swim_intent.py [hunk 11]
```diff
@@ -81,19 +81,20 @@
...
                     type: str
                   is_third_party:
-                    description: ThirdParty flag.
+                    description: Set the boolean value if image is uploaded from third party (Optional).
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:94]: Flag indicates whether the image is uploaded from a third party.

File: plugins/modules/swim_intent.py [hunk 12]
```diff
@@ -81,19 +81,20 @@
...
                     type: bool
                   vendor:
-                    description: Swim Import Via Url's vendor.
+                    description: Name of vendor applicable only for third party image types (Optional).
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:97]: The name of the vendor, that applies only to third-party image types when importing via URL.

File: plugins/modules/swim_intent.py [hunk 13]
```diff
@@ -169,11 +193,24 @@
         type: dict
         suboptions:
           device_role:
-            description: Device Role. Permissible Values ALL, UNKNOWN, ACCESS, BORDER ROUTER,
+            description: Device Role and  permissible Values are ALL, UNKNOWN, ACCESS, BORDER ROUTER,
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:196]: Defines the device role, with permissible values including ALL, UNKNOWN, ACCESS, BORDER ROUTER, DISTRIBUTION, and CORE.

File: plugins/modules/swim_intent.py [hunk 14]
```diff
@@ -169,11 +193,24 @@
...
+                multiple branch offices.
+              DISTRIBUTION - This role represents function as distribution switches or routers in hierarchical network designs. They aggregate traffic
+                from access switches and route it toward the core of the network or toward other distribution switches.
+              CORE - This role typically represents high-capacity switches or routers that form the backbone of the network. They handle large volumes
+                of traffic and provide connectivity between different parts of network, such as connecting distribution switches or
+                providing interconnection between different network segments.
             type: str
           device_family_name:
-            description: Device family name
+            description: Name of the device family(Switches and Hubs etc.)
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:213]: Specify the name of the device family, such as Switches, Hubs, etc.

File: plugins/modules/swim_intent.py [hunk 15]
```diff
@@ -182,10 +219,20 @@
...
             type: bool
           device_upgrade_mode:
-            description: Swim Trigger Activation's deviceUpgradeMode.
+            description: It specifies the mode of upgrade to be applied to the devices having the following values - 'install', 'bundle', and 'currentlyExists'.
+              install - This mode instructs Cisco Catalyst Center to perform a clean installation of the new image on the target devices.
+                When this mode is selected, the existing image on the device is completely replaced with the new image during the upgrade process.
+                This ensures that the device runs only the new image version after the upgrade is completed.
+              bundle - This mode instructs Cisco Catalyst Center bundles the new image with the existing image on the device before initiating
+                the upgrade process.This mode allows for a more efficient upgrade process by preserving the existing image on the device while
+                adding the new image as an additional bundle.After the upgrade, the device can run either the existing image or the new bundled
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:228]: leave space after .

File: plugins/modules/swim_intent.py [hunk 16]
```diff
@@ -182,10 +219,20 @@
...
+                image, depending on the configuration.
+              currentlyExists - This mode instructs Cisco Catalyst Center to checks if the target devices already have the desired image version
+                installed. If image already present on devices, no action is taken and upgrade process is skipped for those devices. This mode
+                is useful for avoiding unnecessary upgrades on devices that already have the correct image version installed, thereby saving time.
             type: str
-          distributeIfNeeded:
-            description: DistributeIfNeeded flag.
+          distribute_if_needed:
+            description: Set the distribute_if_needed flag while activating the swim image.
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:235]: `description: Enable the distribute_if_needed option when activating the SWIM image.`

File: plugins/modules/inventory_workflow_manager.py (hunk 1 above)
Comment (Abhishek-121) [reply plugins/modules/inventory_workflow_manager.py:1771]: added.

File: plugins/modules/swim_intent.py (hunk 2 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:53]: specify local or remote as option in playbook

File: plugins/modules/swim_intent.py (hunk 3 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:60]: added.

File: plugins/modules/swim_intent.py (hunk 4 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:63]: added.

File: plugins/modules/swim_intent.py (hunk 6 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:69]: added.

File: plugins/modules/swim_intent.py (hunk 7 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:72]: added.

File: plugins/modules/swim_intent.py (hunk 8 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:84]: added.

File: plugins/modules/swim_intent.py (hunk 9 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:88]: added.

File: plugins/modules/swim_intent.py (hunk 10 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:91]: added.

File: plugins/modules/swim_intent.py (hunk 12 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:97]: added.

File: plugins/modules/swim_intent.py (hunk 13 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:196]: added.

File: plugins/modules/swim_intent.py (hunk 14 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:213]: added.

File: plugins/modules/swim_intent.py (hunk 15 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:228]: add space.

File: plugins/modules/swim_intent.py (hunk 16 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:235]: added.

File: plugins/modules/inventory_workflow_manager.py (hunk 1 above)
Comment (Abhishek-121) [reply plugins/modules/inventory_workflow_manager.py:1771]: added.

**Review by madhansansel — APPROVED** [review]

**Pull request merged**